### PR TITLE
enable source link

### PR DESCRIPTION
**Pull request by ctaggart**
This is work in progress. I attempted to enable source link for Suave this morning, but couldn't get libuv to compile with Visual Studio 2017, so I opened up https://github.com/libuv/libuv/pull/1283 looking for help. I'll revisit when that is solved.

### CODE REVIEW

**Comment by ademar** [comment]
Between I recently compiled libuv with VS 2017; I just let it upgrade the projects and it just worked.

**Comment by ctaggart** [comment]
Thanks @ademar, given that comment, I got it working and opened a pull request to libuv to [add Visual Studio 2017 support](https://github.com/libuv/libuv/pull/1285) after some build and powershell hacking. Some questions for ya'll. 
- Where do you do official release builds? AppVeyor, Travis CI, other?
- Do you have performance tests or do performance testing?
- The `appveyor.yml` is building and copying the `Debug` build of libuv. Why not the `Release` version? Does it matter?
- Any reason to be on v1.9.1 of libuv and not something newer? Is there a performance test to cover the upgrading impact?

**Comment by haf** [comment]
Hi @ctaggart:

 - release builds are done locally.
 - We tend to use the command-line load generators; ab, wrk, boom and such.
 - Since we do releases locally, we're OK with Debug builds when running tests; better output that way.
 - No, there's no reason for the old version, except that it hasn't been updated.

**Comment by haf** [comment]
Ping

**Comment by haf** [comment]
This should build successfully. Ready to merge?

File: src/Directory.Build.targets [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  <Target Name="EmbedPaketFiles">
+    <ItemGroup>
+      <EmbeddedFiles Include="$([System.IO.Path]::GetFullPath('%(Compile.Identity)'))"
+        Condition="$([System.String]::Copy('%(Identity)').Contains('paket-files'))" />
+    </ItemGroup>
+    <CreateProperty Value="@(EmbeddedFiles)">  
+      <Output TaskParameter="Value" PropertyName="embed" />  
+    </CreateProperty>
+  </Target>
+</Project>
```
Comment (ctaggart) [inline src/Directory.Build.targets]: This entire file can be replaced with simply another PackageReference in `Directory.Build.props`. 
http://www.nuget.org/packages/SourceLink.Embed.PaketFiles/2.1.2
``` xml
<PackageReference Include="SourceLink.Embed.PaketFiles" Version="2.1.2" PrivateAssets="all" />
```

File: appveyor.yml [hunk 2]
```diff
@@ -8,11 +11,16 @@ environment:
      MSBUILD_PLATFORM: "Any CPU"
 
 install:
- - ps: Start-FileDownload 'https://github.com/libuv/libuv/archive/v1.9.1.zip'
- - cmd: 7z x v1.9.1.zip & cd libuv-1.9.1 & vcbuild.bat %arch% shared debug
- - cmd: mkdir src\Suave.Tests\bin\Release\ & cp libuv-1.9.1\Debug\libuv.dll src\Suave.Tests\bin\Release\libuv.dll
+ - ps: Start-FileDownload 'https://github.com/libuv/libuv/archive/v1.11.0.zip'
+ - cmd: |
+        "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat"
+        set GYP_MSVS_VERSION=2015
```
Comment (ctaggart) [inline appveyor.yml]: Not sure if GYP_MSVS_VERSION is needed given the fixes for libuv. Hoping not.

**Comment by ctaggart** [comment]
@haf, I'm not sure if you want to wait for [official VS 2017 support from libuv](https://github.com/libuv/libuv/pull/1284) or not. The decisions you'll need to make are embedded pdb or just regular portable pdb files. If you go with regular portable pdb files, you will need to include them in the nupkg's. That appears to be what most of choosing to do.

**Comment by haf** [comment]
I'd say, let's go with the flow and use portable :)

**Comment by haf** [comment]
Ready to merge?

**Comment by haf** [comment]
Ping @ctaggart

**Comment by ctaggart** [comment]
@haf, this code looks good. Are you already packaging the pdb files in the nupkg?

File: src/Directory.Build.props [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<Project>
+  <PropertyGroup>
+    <DebugType Condition="'$(DebugType)' == ''">embedded</DebugType>
```
Comment (ctaggart) [inline src/Directory.Build.props]: This should be switched to portable.

**Comment by haf** [comment]
Yeah, I'm ready, but we should decide whether we want to keep the existing infrastructure to do that. We currently are merging nugets/nupkg files with @enricosada 's code.

**Comment by ctaggart** [comment]
So, I don't have the time to see this PR through to the finish right now. Should I close it?

**Comment by haf** [comment]
@ctaggart No need to close it. Finish it when you have time available; we greatly appreciate your contributions.

I just just checking you for a heartbeat over there... ;)

File: paket.lock [hunk 4]
```diff
@@ -186,7 +201,7 @@ NUGET
       Expecto (>= 4.0.3)
       FsCheck (>= 2.7.1)
       FSharp.Core (>= 3.1.2.5)
-    FsCheck (2.7.1)
+    FsCheck (2.8.2)
```
Comment (ctaggart) [inline paket.lock]: oops

**Comment by ctaggart** [comment]
This is working:
```
C:\Users\camer\cs\SourceLink\build [embed-commmandline ≡]> dotnet sourcelink test "C:\Users\camer\Downloads\Suave.2.3.0-beta2(1)\lib\netstandard2.0\Suave.pdb"
sourcelink test passed: C:\Users\camer\Downloads\Suave.2.3.0-beta2(1)\lib\netstandard2.0\Suave.pdb
```

Please merge #679 first. I'm release a stable SourceLink 2.7.2 now.

**Pull request merged**